### DISCUSSION TITLE
I am unable to run the exe file pyinstaller creates from a very simple test.py program and am grateful for any help

### DISCUSSION THREAD

**Discussion by jgsmcmahon** [pyinstaller/pyinstaller#7405] [Help]
I have set the debugging on and get this message when I try and run the exe that is created.. I am running on Ubuntu any clues to whats going on much appreciated :-) 

Error processing line 1 of /usr/lib/python3/dist-packages/zope.interface-4.7.1-nspkg.pth:

  Traceback (most recent call last):
    File "/usr/lib/python3.8/site.py", line 175, in addpackage
      exec(line)
    File "<string>", line 1, in <module>
  ImportError: bad magic number in 'importlib': b'B\r\r\n'

Remainder of file ignored
Traceback (most recent call last):
  File "/usr/lib/command-not-found", line 23, in <module>
    import gettext
ImportError: bad magic number in 'gettext': b'B\r\r\n'
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 72, in apport_excepthook
    from apport.fileutils import likely_packaged, get_recent_crashes
  File "/usr/lib/python3/dist-packages/apport/__init__.py", line 5, in <module>
    from apport.report import Report
  File "/usr/lib/python3/dist-packages/apport/report.py", line 12, in <module>
    import subprocess, tempfile, os.path, re, pwd, grp, os, time, io
ImportError: bad magic number in 'subprocess': b'B\r\r\n'

Original exception was:
Traceback (most recent call last):
  File "/usr/lib/command-not-found", line 23, in <module>
    import gettext
ImportError: bad magic number in 'gettext': b'B\r\r\n'

**Comment by rokm** [comment]
`'B\r\r\n'` is the bytecode magic number for python 3.7. 

The most recent traceback indicates that python 3.8 is in fact used in that environment, but I think this exception is coming from system-installed `apport` service trying to analyze the issue. But then again, the original exception was also triggered by stdlib module not being loaded due to magic number mismatch. 

So my guess is that somehow modules from incompatible python version (3.7) were collected into your frozen application, but you have provided insufficient information about your environment for us to guess as to how and why.

**Reply by jgsmcmahon** [reply]
Rokm 
Much appreciate your answer I have several versions of python installed and wonder if my paths are getting messed up. I am self taught and suspect that I am not setting the environment up correctly. I have been using Jupyter notebooks mainly and have set virtual envs using anancoda conda. Does this help ? Kind regards 

~$ whereis python
python: /usr/bin/python2.7-config /usr/bin/python2.7 /usr/bin/python3.8 /usr/bin/python /usr/lib/python2.7 /usr/lib/python3.8 /usr/lib/python3.9 /etc/python3.6 /etc/python2.7 /etc/python3.8 /etc/python /usr/local/lib/python2.7 /usr/local/lib/python3.8 /usr/include/python2.7 /usr/include/python3.8 /usr/share/python /home/jgsmcmahon/anaconda3/bin/python3.7m-config /home/jgsmcmahon/anaconda3/bin/python3.7-config /home/jgsmcmahon/anaconda3/bin/python3.7m /home/jgsmcmahon/anaconda3/bin/python3.7 /home/jgsmcmahon/anaconda3/bin/python

**Reply by jgsmcmahon** [reply]
Also this is the message string when I try and make the exe, which may shed some light on this ... 

Running auto-py-to-exe v2.27.0
Building directory: /tmp/tmprnuvq01e
Provided command: pyinstaller --noconfirm --onedir --console --debug "all"  "/home/jgsmcmahon/Desktop/Python/metascan/sandbox/jason.py"
Recursion Limit is set to 5000
Executing: pyinstaller --noconfirm --onedir --console --debug all /home/jgsmcmahon/Desktop/Python/metascan/sandbox/jason.py --distpath /tmp/tmprnuvq01e/application --workpath /tmp/tmprnuvq01e/build --specpath /tmp/tmprnuvq01e

1090153 INFO: PyInstaller: 5.7.0
1090155 INFO: Python: 3.7.16 (conda)
1090157 INFO: Platform: Linux-5.4.0-131-generic-x86_64-with-debian-bullseye-sid
1090160 INFO: wrote /tmp/tmprnuvq01e/jason.spec
1090166 INFO: UPX is not available.
1090171 INFO: Extending PYTHONPATH with paths
['/home/jgsmcmahon/Desktop/Python/metascan/sandbox']
1090355 INFO: checking Analysis
1090356 INFO: Building Analysis because Analysis-03.toc is non existent
1090358 INFO: Reusing cached module dependency graph...
1090384 INFO: Caching module graph hooks...
1090452 INFO: running Analysis Analysis-03.toc
1090461 INFO: Analyzing /home/jgsmcmahon/Desktop/Python/metascan/sandbox/jason.py
1090463 INFO: Processing module hooks...
1090465 INFO: Loading module hook 'hook-pickle.py' from '/home/jgsmcmahon/anaconda3/envs/metascan/lib/python3.7/site-packages/PyInstaller/hooks'...
1090468 INFO: Loading module hook 'hook-heapq.py' from '/home/jgsmcmahon/anaconda3/envs/metascan/lib/python3.7/site-packages/PyInstaller/hooks'...
1090470 INFO: Loading module hook 'hook-encodings.py' from '/home/jgsmcmahon/anaconda3/envs/metascan/lib/python3.7/site-packages/PyInstaller/hooks'...
1090723 INFO: Looking for ctypes DLLs
1090725 INFO: Analyzing run-time hooks ...
1090731 INFO: Looking for dynamic libraries
1091429 INFO: Looking for eggs
1091432 INFO: Python library not in binary dependencies. Doing additional searching...
1091442 INFO: Using Python library /home/jgsmcmahon/anaconda3/envs/metascan/lib/libpython3.7m.so.1.0
1091447 INFO: Warnings written to /tmp/tmprnuvq01e/build/jason/warn-jason.txt
1091465 INFO: Graph cross-reference written to /tmp/tmprnuvq01e/build/jason/xref-jason.html
1091472 INFO: checking PYZ
1091475 INFO: Building PYZ because PYZ-03.toc is non existent
1091479 INFO: Building PYZ (ZlibArchive) /tmp/tmprnuvq01e/build/jason/PYZ-03.pyz
1091597 INFO: Building PYZ (ZlibArchive) /tmp/tmprnuvq01e/build/jason/PYZ-03.pyz completed successfully.
1091600 INFO: checking PKG
1091602 INFO: Building PKG because PKG-03.toc is non existent
1091603 INFO: Building PKG (CArchive) jason.pkg
1091616 INFO: Building PKG (CArchive) jason.pkg completed successfully.
1091619 INFO: Bootloader /home/jgsmcmahon/anaconda3/envs/metascan/lib/python3.7/site-packages/PyInstaller/bootloader/Linux-64bit-intel/run_d
1091620 INFO: checking EXE
1091622 INFO: Building EXE because EXE-03.toc is non existent
1091625 INFO: Building EXE from EXE-03.toc
1091626 INFO: Copying bootloader EXE to /tmp/tmprnuvq01e/build/jason/jason
1091628 INFO: Appending PKG archive to custom ELF section in EXE
1091634 INFO: Building EXE from EXE-03.toc completed successfully.
1091637 INFO: checking COLLECT
1091639 INFO: Building COLLECT because COLLECT-02.toc is non existent
1091640 INFO: Building COLLECT COLLECT-02.toc
1091659 INFO: Building COLLECT COLLECT-02.toc completed successfully.

**Reply by rokm** [reply]
Ah, the original exception, 
```
Original exception was:
Traceback (most recent call last):
File "/usr/lib/command-not-found", line 23, in
import gettext
ImportError: bad magic number in 'gettext': b'B\r\r\n'
```
is also coming from system-installed program/utility `/usr/lib/command-not-found`. So what is probably happening is that `apport` is ran (maybe your actual program terminates with unhandled exception, which triggers `apport` processing?) and the .pyc files from your PyInstaller-frozen application (python 3.7) are being loaded by system python 3.8 interpreter.

Avoid using `--debug all`, which implies `--debug noarchive` and causes the collected modules to be written as .pyc files instead of being collected into PYZ archive (i.e., use `--debug bootloader`  or `--debug imports`, depending on what you want to achieve). 

Or disable `apport` while you are debugging.
